src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj

```diff
@@ -72,16 +72,28 @@
     * :http-client-cipher-suites - A list of legal SSL cipher suites that may
         be used when https client requests are made.
 
+    * :http-client-connect-timeout - The amount of time, in milliseconds, that
+        an outbound HTTP connection will wait for no data to be available before
+        closing the socket. Defaults to 2 minutes. If 0, the timeout is infinite
```

Comment:
This description of `:http-client-connect-timeout` sounds like it's describing the behavior of the timeout associated with a connected socket that's idling, but I think that's the behavior of `:http-client-socket-timeout` unless I'm mistaken.